### ISSUE TITLE
[clangd] False "unused-includes" for types used only as explicit template arguments of deducing this

### ISSUE THREAD

**Issue by AkrinW** [llvm/llvm-project#218829]
## Summary
clangd reports a false positive "unused include" when a type from a header is used only as an explicit template argument of a call to a "deducing this". Removing the include as suggested breaks the build.
  Tested with clangd 23.1.0.

## Minimal reproducer
`foo.h`:
```cpp
#pragma once

struct Foo {};
```

`mre.cpp`:
```cpp
#include "foo.h"
struct S {
    template <typename T>
    void f(this S &);
};
void test(S &s) {
    s.f<Foo>();
}
```
Commands:
```console
$ clang++ -std=c++23 -fsyntax-only mre.cpp          # compiles fine
$ clang-include-cleaner --print=changes --disable-insert mre.cpp
- "foo.h" @Line:1
```
## Expected behavior
`foo.h` must be kept: clangd should mark `Foo` as used, exactly as it does for the equivalent imp:
```cpp
template <typename T>
void f();          // no explicit object parameter -> no false positive
```

 ## Root cause analysis
**Generated by Claude**

  When Sema builds a call to an explicit object member function, it lowers the member access into a plain `CallExpr` whose callee is a `DeclRefExpr` (`Sema::BuildCallToMemberFunction`, `clang/lib/Sema/SemaOverload.cpp`). That `DeclRefExpr` is created via `CreateFunctionRefExpr`, which does **not** pass a `TemplateArgumentListInfo`, so the explicit template arguments written in the source (`<Foo>`) are dropped from the AST. For the reproducer the AST is just:
  CallExpr
  ├── DeclRefExpr 'f'      <-- no explicit template arguments attached
  └── DeclRefExpr 's'

  include-cleaner walks explicit template arguments of `DeclRefExpr` / `MemberExpr` in `clang-tools-extra/include-cleaner/lib/WalkAST.cpp` (`VisitDeclRefExpr` → template argument traversal), but since the lowered `DeclRefExpr` carries no `TemplateArgumentLoc`s, `Foo` is never reported as a use, and `foo.h` is flagged as unused. Note this is the same lowering that caused #116928 (wrong begin location of the resulting `CallExpr`).
The fix there only adjusted source locations; the explicit template arguments are still not preserved on the synthesized `DeclRefExpr`.
